main.go

```diff
@@ -0,0 +1,193 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-tools/go-steputils/stepconf"
+	semver "github.com/hashicorp/go-version"
+)
+
+// Config model
+type Config struct {
+	Workdir    string `env:"workdir"`
+	Command    string `env:"command,required"`
+	NpmVersion string `env:"npm_version"`
+}
+
+func getNpmVersionFromPackageJSON() (string, error) {
+	jsonStr, err := fileutil.ReadStringFromFile("package.json")
+	if err != nil {
+		return "", fmt.Errorf("package.json file read error: %s", err)
+	}
+
+	var ver string
+	if ver, err = extractNpmVersion(jsonStr); err != nil {
+		return "", fmt.Errorf("package json content error: %s", err)
+	}
+	return ver, nil
+}
+
+func extractNpmVersion(jsonStr string) (string, error) {
+	type jsonModel struct {
+		Engines struct {
+			Npm string
+		}
+	}
+
+	var m jsonModel
+	if err := json.Unmarshal([]byte(jsonStr), &m); err != nil {
+		return "", fmt.Errorf("json unmarshal error: %s", err)
+	}
+
+	if m.Engines.Npm == "" {
+		return "", nil
+	}
+
+	v, err := semver.NewVersion(m.Engines.Npm)
+	if err != nil {
+		return "", fmt.Errorf("engines.npm is not valid semver string")
+	}
+
+	return v.String(), nil
+}
+
+func createInstallNpmCommand(os string) *command.Model {
+	var args []string
+	switch os {
+	case "darwin":
+		args = []string{"brew", "install", "node"}
+	case "linux":
+		args = []string{"apt-get", "-y", "install", "npm"}
+	}
+
+	return command.New(args[0], args[1:]...)
+}
+
+func installLatestNpm() (string, error) {
+	cmd := createInstallNpmCommand(runtime.GOOS)
+
+	var out string
+	out, err := command.RunCmdAndReturnTrimmedOutput(cmd.GetCmd())
+
+	if err != nil {
+		return out, fmt.Errorf("error running npm install: %s", err)
+	}
+
+	return out, nil
+}
+
+func runAndLog(cmd *command.Model) (string, error) {
+	out, err := cmd.RunAndReturnTrimmedCombinedOutput()
+	if err != nil {
+		return out, fmt.Errorf("error running npm command: %s", err)
+	}
+
+	log.Donef(fmt.Sprintf("$ %s", cmd.PrintableCommandArgs()))
```

Comment:
print the cmd before it would be performed, otherwise in case of error we will not know what command failed